packages/edit-site/src/components/sidebar-navigation-screen/index.js

```diff
@@ -29,7 +28,26 @@ import {
 	currentlyPreviewingTheme,
 } from '../../utils/is-previewing-theme';
 
-const { useLocation } = unlock( routerPrivateApis );
+const { useHistory, useLocation } = unlock( routerPrivateApis );
+
+function getBackPath( params ) {
+	// From a data view path we navigate back to root
+	if ( params.path ) {
+		return {};
+	}
+
+	// From edit screen for a post we navigate back to post-type specific data view
+	if ( params.postType === 'page' ) {
+		return { path: '/page', postId: params.postId };
+	} else if ( params.postType === 'wp_template' ) {
+		return { path: '/wp_template', postId: params.postId };
+	} else if ( params.postType === 'wp_navigation' ) {
+		return { path: '/navigation', postId: params.postId };
+	}
+
+	// Go back to root by default
+	return {};
```

Comment:
I'm not sure that's always true for non post type screens? It might be the case now but I don't feel like this is the right path for a generic solution. I feel like maybe the best way to solve this is to actually pass the right "back path" on each sidebar component.

Obviously not something that need to be solved now or anything, just thinking out loud.

Reply:
I think it's correct for all existing screens. I'm currently not aware of any navigation link that would point to a wrong direction.

But yes, I also think we'll end up passing an explicit `backPath` to every screen, instead of trying to guess it with some heuristic.

Reply:
Agree with this follow-up as well.